packages/core/src/schema/blocks/playground.ts

```diff
@@ -0,0 +1,204 @@
+
+export type BlockDefs = Record<string, BlockConfig<string, PropSchema>>;
+
+export interface BlockConfigMeta {
+  /**
+   * Whether the block is selectable
+   */
+  selectable?: boolean;
+
+  /**
+   * The accept mime types for the file block
+   */
+  fileBlockAccept?: string[];
+
+  /**
+   * Whether the block is a {@link https://prosemirror.net/docs/ref/#model.NodeSpec.code} block
+   */
+  code?: boolean;
+
+  /**
+   * Whether the block is a {@link https://prosemirror.net/docs/ref/#model.NodeSpec.defining} block
+   */
+  defining?: boolean;
+}
+
+/**
+ * BlockConfig contains the "schema" info about a Block type
+ * i.e. what props it supports, what content it supports, etc.
+ */
+export interface BlockConfig<
+  TName extends string,
+  TSchema extends PropSchema = PropSchema,
+> {
+  /**
+   * The type of the block (unique identifier within a schema)
+   */
+  type: TName;
+  /**
+   * The properties that the block supports
+   * @todo will be zod schema in the future
+   */
+  readonly propSchema: TSchema;
+  /**
+   * The content that the block supports
+   */
+  content: "inline" | "none";
+  // TODO: how do you represent things that have nested content?
+  // e.g. tables, alerts (with title & content)
+  /**
+   * Metadata
+   */
+  meta?: BlockConfigMeta;
+}
+
+export interface BlockImplementation<
+  TName extends string,
+  TProps extends PropSchema,
+> {
+  /**
+   * A function that converts the block into a DOM element
+   */
+  render: (
+    /**
+     * The custom block to render
+     */
+    block: Block<Record<TName, BlockConfig<TName, TProps>>>,
+    /**
+     * The BlockNote editor instance
+     */
+    editor: BlockNoteEditor<Record<TName, BlockConfig<TName, TProps>>>,
+  ) => {
+    dom: HTMLElement | DocumentFragment;
+    contentDOM?: HTMLElement;
+    ignoreMutation?: (mutation: ViewMutationRecord) => boolean;
+    destroy?: () => void;
+  };
+
+  /**
+   * Exports block to external HTML. If not defined, the output will be the same
+   * as `render(...).dom`.
+   */
+  toExternalHTML?: (
+    block: Block<Record<TName, BlockConfig<TName, TProps>>>,
+    editor: BlockNoteEditor<Record<TName, BlockConfig<TName, TProps>>>,
+  ) =>
+    | {
+        dom: HTMLElement;
+        contentDOM?: HTMLElement;
+      }
+    | undefined;
+
+  /**
+   * Parses an external HTML element into a block of this type when it returns the block props object, otherwise undefined
+   */
+  parse?: (el: HTMLElement) => NoInfer<Partial<Props<TProps>>> | undefined;
+
+  /**
+   * The blocks that this block should run before.
+   * This is used to determine the order in which blocks are rendered.
+   */
+  runsBefore?: string[];
+
+  /**
+   * Advanced parsing function that controls how content within the block is parsed.
+   * This is not recommended to use, and is only useful for advanced use cases.
+   */
+  parseContent?: (options: { el: HTMLElement; schema: Schema }) => Fragment;
```

Comment:
would be nice to do this in a way that we don't expose Prosemirror APIs (i.e.: not expose Fragment) - not sure this is possible